pride/src/integTest/groovy/com/prezi/pride/test/HgCacheRepoTest.groovy

```diff
@@ -0,0 +1,34 @@
+package com.prezi.pride.test
+
+class HgCacheRepoTest extends AbstractIntegrationSpec implements HgSupport {
+	def "stale repo should be reinitialized if found stale"() {
+		given:
+		initializeHgRepo('project')
+		ensurePrideModuleIsCached('project')
+		makeModuleRepoUnrelatedToTheOneInPrideCache('project')
+
+		when:
+		initializePrideWithModule(cache: true, 'pride', 'project')
+
+		then:
+		notThrown(Exception)
+
+		when:
+		commitNewEmptyFile('project', 'new-file')
+		pride workingDir: 'pride', 'update'
+
+		then:
+		file('pride/project/new-file').isFile()
+	}
+
+	private void ensurePrideModuleIsCached(String moduleName) {
+		def tempWorkingDir = file('temporary-pride-working-directory-used-to-init-cache')
+		initializePrideWithModule(cache: true, tempWorkingDir, moduleName)
+		assert tempWorkingDir.deleteDir()
```

Comment:
This should assert that the directory exists, and delete it in a `finally { }` block.